### ISSUE TITLE
Using cflags in manifest file broken in latest build

### ISSUE THREAD

**Issue by tonis2** [c3lang/c3c#1369]
Previously I could use `cflags` attribute in **manifest.json**

For example I used below attribute in a library, and it was working fine

```
"cflags": "-fPIE -D RGFW_NO_X11_CURSOR -D RGFW_VULKAN_IMPLEMENTATION -D RGFW_NO_API -D RGFW_IMPLEMENTATION -D RGFW_NO_X11_CURSOR",
```

But with latest builds, the compiler started adding `\`  breaks between values, and now I am getting below error

```
cc: error: unrecognized command-line option ‘-fPIE -D RGFW_NO_X11_CURSOR -D RGFW_VULKAN_IMPLEMENTATION -D RGFW_NO_API -D RGFW_IMPLEMENTATION -D RGFW_NO_X11_CURSOR

Failed to compile c sources using command 'cc -c -fPIE\ -D\ RGFW_NO_X11_CURSOR\ -D\ RGFW_VULKAN_IMPLEMENTATION\ -D\ RGFW_NO_API\ -D\ RGFW_IMPLEMENTATION\ -D\ RGFW_NO_X11_CURSOR ./rgfw.c3l/rgfw.c -o build/tmp/rgfw/rgfw.o'.
```

I think, the issue is caused by these added `\`  values between -D flags

**Comment by lerno** [comment]
I think maybe @cbuttner made this change, could you have a look?

**Comment by lerno** [comment]
I'm reworking this a bit. Expect a version tomorrow.

**Comment by lerno** [comment]
This should now work.

**Comment by cbuttner** [comment]
The fix would've been to make cflags an array like linker args, not to remove my escaping code which worked fine.

**Comment by lerno** [comment]
Take a look at how I changed it. I've altered the way things are rendered which also circumvents the need for strformat

**Comment by cbuttner** [comment]
Yes I looked at it, and you're overcomplicating this.

It's simple: Have one string equal one arg, and escape if needed before it's passed to the command line. I tested this, and I consider it to work unless it's proven not to.
Instead you added three extra functions that make unnecessary distinctions.

Also the process name is not an argument, it must be handled differently. If you quote the process name, unless it's a full path, it won't run on windows command line. So the actual robust way is to take the executable name or path and a list of arguments assembled into the command line and then call `CreateProcessW` instead of `system`.

Also `"` or `\` is no longer escaped properly. As an example, when a quoted argument ends with `\`, it'll escape the ending `"` away. There's a good reason why things are the way they are, it's all intentional. I don't understand why you changed it.

Not to mention there's other code which also direly in need of correct argument handling. For example you're running `rm -f` with unescaped arguments, which can lead to a very bad time.

**Comment by lerno** [comment]
The reason I changed it was because:
1. It did a lot of unnecessary strformat (in my old code)
2. It relied on `\ ` to escape, which isn't guaranteed to work on Windows for escaping paths
3. It didn't work for user provided flags
4. There were unsafe uses of scratch buffer, which would overwrite it.
5. I looked at the code and realized that the proper way to do this would be to make the assembling function aware of the arguments.

I wanted a solution where it was clear when assembling each argument what it was supposed to require. Note how `""` isn't even the required strategy. In the changed code it does indeed do `""` when it reaches something that needs to be quoted BUT this is not hard coded. So it is in fact possible to instead render a quoted arguments with escaped spaces etc, leaving it open for further such improvements.
